src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -238,25 +255,62 @@ public void setExporter(final Settings settings) {
     }
 
     /**
-     * Validate provided settings for top queries exporter
+     * Set up the top queries reader based on provided settings
      *
-     * @param settings settings exporter config {@link Settings}
+     * @param settings reader config {@link Settings}
+     * @param namedXContentRegistry NamedXContentRegistry for parsing purposes
      */
-    public void validateExporterConfig(Settings settings) {
+    public void setReader(final Settings settings, final NamedXContentRegistry namedXContentRegistry) {
+        this.reader = queryInsightsReaderFactory.createReader(
+            settings.get(EXPORT_INDEX, DEFAULT_TOP_N_QUERIES_INDEX_PATTERN),
+            namedXContentRegistry
+        );
+        queryInsightsReaderFactory.updateReader(reader, settings.get(EXPORT_INDEX, DEFAULT_TOP_N_QUERIES_INDEX_PATTERN));
+    }
+
+    /**
+     * Validate provided settings for top queries exporter and reader
+     *
+     * @param settings settings exporter/reader config {@link Settings}
+     */
+    public void validateExporterReaderConfig(Settings settings) {
         queryInsightsExporterFactory.validateExporterConfig(settings);
+        queryInsightsReaderFactory.validateReaderConfig(settings);
     }
 
+    /**
+     * Lambda function to mark if a record is internal
+     */
+    private final Predicate<SearchQueryRecord> removeInternal = (record) -> {
```

Comment:
This lambda function is more about "check if is internal", maybe let's rename it to 
```
checkIfInternal
```
?